test/test_dbgfile.py

```diff
@@ -0,0 +1,40 @@
+import unittest
+import os
+
+from elftools.elf.elffile import ELFFile, DynamicSection
+from elftools.dwarf.callframe import ZERO
+
+class TestDBGFile(unittest.TestCase):
+    def test_dynamic_segment(self):
+        with open(os.path.join('test', 'testfiles_for_unittests',
+                               'debug_info.elf'), 'rb') as f:
+            elf = ELFFile(f)
+            
+            for segment in elf.iter_segments():
```

Comment:
I think this loop would be more readable using more standard means. Having to remember the interaction between `break` and `for...else` is too much for my brain capacity :-(

```
go over all segments in file
if segment is PT_DYNAMIC:
  seen_dynamic_segment = true
  fail if num_tasg != 0
fail if !seen_dynamic_segment
```
```

Reply:
unfortunately, I've been writing so much python lately at the for...else is more readable to me :-(